virttest/env_process.py

```diff
@@ -1011,15 +1011,24 @@ def preprocess(test, params, env):
             test.cancel("Failed to map hostname and ipaddress of target host")
         session.close()
 
-    # Destroy and remove VMs that are no longer needed in the environment
+    # Destroy and remove VMs that are no longer needed in the environment or
+    # leave them untouched if they have to be disregarded only for this test
     requested_vms = params.objects("vms")
+    vm_env_action = params.get("vm_env_action", "destroy")
```

Comment:
I'd suggest separating this feature into a new commit.

Reply:
If you suggest a separate commit, then I would argue it makes more sense in the commit about making vms persistent which includes the optional cleanup of the env file and the optional destruction of vm objects. If you mean it as a separate PR, then I think this PR is starting to turn into a PR about making the `tmp_dir` permanent which was never its focal point and while this is a good addition, I prefer the PR to remain focused on the vm permanence. An extra commit for the persistent tmp directory should be enough.